pymeasure/instruments/instrument.py

```diff
@@ -81,15 +84,21 @@ def id(self):
             return self.adapter.ask("*IDN?").strip()
         else:
             return "Warning: Property not implemented."
+    
+    @property
+    def id_dict(self):
+        return dict(zip(("vendor", "model", "serial", "firmware"), self.id.split(",")))
 
     # Wrapper functions for the Adapter object
-    def ask(self, command):
+    def query(self, command):
         """ Writes the command to the instrument through the adapter
         and returns the read response.
 
         :param command: command string to be sent to the instrument
         """
         return self.adapter.ask(command)
+    
+    ask = query # ask is deprecated
```

Comment:
This is deprecated by whom? can you clarify this? also, note that in the line above you still do adapter.ask...

Reply:
by pyvisa, if I remember right.

Reply:
I checked, this was deprecated in [pyvisa 1.9](https://github.com/pyvisa/pyvisa/blob/0147607bccfb1c7de9d44d36d66ec543d93e8b3e/CHANGES#L93) and will be [removed in 1.10](https://github.com/pyvisa/pyvisa/blob/47cc2b1ae3318b7b437cc7a49be8778c713377b8/pyvisa/resources/messagebased.py#L615) (although at 1.10.1, I still have the warning), so thank you for spotting this!
However, this has already been fixed in the [visa adapter](https://github.com/ralph-group/pymeasure/blob/master/pymeasure/adapters/visa.py#L112), so there is no need to change our own public API.
(As an aside, if you end up deprecating something, it's better to notify users about this, e.g. like [this](https://github.com/pyvisa/pyvisa/blob/47cc2b1ae3318b7b437cc7a49be8778c713377b8/pyvisa/resources/messagebased.py#L615-L618))

Reply:
The purpose of the `Adapter` classes is to abstract away the details of the underlying communication protocols/libraries. We originally based the API on an earlier version of pyvisa, but we are not tied the pyvisa API. The `Instrument` API is stable, and is decoupled from the specific adapters. As @bilderbuchi points out, changes were made specifically to the VISAAdapter to handle changes based on the version of pyvisa.

Reply:
Yes, but when porting code from using pyvisa to using pymeasure it'd be nice to have methods (that are in fact aliases) mimicking current pyvisa API. Since it doesn't harm, I have added an alias.